CPAC/pipeline/nodeblock.py

```diff
@@ -0,0 +1,115 @@
+from typing import Callable, List, Union, Optional, Dict, Any
+
+
+class NodeBlockFunction(object):
+    """
+    Stores a reference to the nodeblock function and all of its meta-data.
+    """
+
+    def __init__(
+            self,
+            func: Callable,
+            name: Optional[str] = None,
+            config: Optional[List[str]] = None,
+            switch: Optional[Union[List[str], List[List[str]]]] = None,
+            option_key: Optional[Union[str, List[str]]] = None,
+            option_val: Optional[Union[str, List[str]]] = None,
+            inputs: Optional[List[Union[str, list, tuple]]] = None,
+            outputs: Optional[Union[List[str], Dict[str, Any]]] = None
+    ) -> None:
+        self.func = func
+        """Nodeblock function reference."""
```

Comment:
This is pedantic, but the codebase mostly follows PEP8's comment format recommendations:
triple-quotes for docstrings, `#` for [block comments (preceding relevant code)](https://peps.python.org/pep-0008/#block-comments) and inline comments.

So like
```Python
        # Nodeblock function reference.
        self.func = func
```
instead of
```Python
        self.func = func
        """Nodeblock function reference."""
```
and
```Python
        # Indicates the nested keys in a C-PAC pipeline configuration
        # should configure a NodeBlock built from this function. If
        # config is set to "None", then all other configuration-related
        # entities must be specified from the root of the configuration.
        self.config: Optional[List[str]] = config
```
instead of
```Python
        self.config: Optional[List[str]] = config
        """
        Indicates the nested keys in a C-PAC pipeline configuration should configure a NodeBlock built from this
        function. If config is sct to "None", then all other configuration-related entities must be specified from the
        root of the configuration.
        """
```

Unless there's a reason to comment differently here, I think consistency is good to aim for.

Reply:
These are Docstrings for the different object fields. Docstrings are `"""` in the guide you linked. (This way the string gets displayed in modern IDEs when you hover over `some_nodeblock_fun.switch`)

Reply:
![image](https://github.com/FCP-INDI/C-PAC/assets/33600480/318b1c43-7751-47e2-871a-f5b2bc75a9d0)

vs

![image](https://github.com/FCP-INDI/C-PAC/assets/33600480/71afedec-60b6-45f2-a61b-f8d4a0b5701b)

Reply:
Woah cool